File: apps/ios/Barns/BarnsApp/Domain/UseCases/Home/GetHomeSummaryUseCase.swift
/// Provides the lightweight home summary.
struct GetHomeSummaryUseCase {
    private let homeRepository: HomeRepository

    init(homeRepository: HomeRepository) {
        self.homeRepository = homeRepository
    }

    func execute() async throws -> HomeSummary {
        try await homeRepository.homeSummary()
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/home/GetHomeSummaryUseCase.kt
package com.barns.app.domain.usecase.home

import com.barns.app.domain.model.HomeSummary
import com.barns.app.domain.repository.HomeRepository

/**
 * Provides the lightweight home summary.
 */
class GetHomeSummaryUseCase(
    private val homeRepository: HomeRepository,
) {
    suspend fun execute(): HomeSummary = homeRepository.homeSummary()
}
